src/main/webapp/app/more-feedback/more-feedback.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JhiAlertService } from 'ng-jhipster';
+import { ComplaintService } from 'app/entities/complaint/complaint.service';
+import { Complaint, ComplaintType } from 'app/entities/complaint';
+import { Result } from 'app/entities/result';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Moment } from 'moment';
+import { ComplaintResponseService } from 'app/entities/complaint-response/complaint-response.service';
+import { ComplaintResponse } from 'app/entities/complaint-response';
+
+@Component({
+    selector: 'jhi-more-feedback-form',
+    templateUrl: './more-feedback.component.html',
+    providers: [JhiAlertService],
+})
+export class MoreFeedbackComponent implements OnInit {
+    @Input() resultId: number;
+    complaintText = '';
```

Comment:
Most of the attributes in this component seem to me like they are properties of a complaint / feedback response.
If this is the case, than it would be better imo to keep them in a class / interface / type object. Otherwise it is harder to reason about the state of the component (what if alreadySubmitted is true but handled is false?).

Reply:
We decided not to change backend implementation of complaint